src/Microsoft.OpenApi/Models/OpenApiSchema.cs

```diff
@@ -549,7 +582,15 @@ internal void WriteAsSchemaProperties(IOpenApiWriter writer)
             writer.WriteProperty(OpenApiConstants.Discriminator, Discriminator?.PropertyName);
 
             // readOnly
-            writer.WriteProperty(OpenApiConstants.ReadOnly, ReadOnly, false);
+            // In V2 schema if a property is part of required properties of schema, it cannot be marked as readonly.
+            if (parentRequiredProperties.Contains(propertyName))
+            {
+                writer.WriteProperty(OpenApiConstants.ReadOnly, false, false);
```

Comment:
>         writer.WriteProperty(OpenApiConstants.ReadOnly, false, false); [](start = 7, length = 71)

Let's just not write anything since the default value according to the spec is already false.